src/consumers/node-stream.js

```diff
@@ -16,23 +16,23 @@ function toNodeStream (renderer) {
 
   const read = () => {
     // If source is not ready, defer any reads until the promise resolves.
-    if (!sourceIsReady) { return; }
+    if (!sourceIsReady) { return false; }
 
     sourceIsReady = false;
     const pull = pullBatch(renderer, stream);
 
-    pull.then(result => {
+    return pull.then(result => {
       sourceIsReady = true;
       if (result === EXHAUSTED) {
-        stream.push(null);
+        return stream.push(null);
```

Comment:
I understand why we need to return the promise above, but why do we have to to return stream `.push` and `.emit`, and `read`?